be/src/storage/lake/meta_file.cpp

```diff
@@ -56,13 +56,34 @@ void MetaFileBuilder::append_delvec(const DelVectorPtr& delvec, uint32_t segment
 
 void MetaFileBuilder::apply_opwrite(const TxnLogPB_OpWrite& op_write,
                                     const std::map<int, std::string>& replace_segments,
-                                    const std::vector<std::string>& orphan_files) {
+                                    const std::vector<std::string>& orphan_files,
+                                    std::map<std::string, uint64_t>& replace_segments_file_size) {
     auto rowset = _tablet_meta->add_rowsets();
     rowset->CopyFrom(op_write.rowset());
+
+    bool upgrage_from_old_version = rowset->segment_size_size() == 0;
```

Comment:
```suggestion
    bool upgrage_from_old_version = rowset->segment_size_size() == rowset->segments_size();
```

Reply:
As a form of defensive programming against having unexpected segment sizes and segment names that are not aligned